src/main/thrift/parquet.thrift

```diff
@@ -174,7 +174,15 @@ enum ConvertedType {
    * particular timezone or date.
    */
   INTERVAL = 21;
-  
+
+  /**
+   * A Union type
+   * This type annotates data stored as a Group
+   * this shows the intent to have heterogenous types under the same field name
+   * the names of the fields in the annotated Group are not important in such a case
```

Comment:
I think this should include a few more details for implementing the Union type. For example, all fields of a union group must be optional and for each value, writers must set all but one option/field to null. It should also state that if more than one option is set, only one will be returned and which one is implementation-specific (because this is affected by column projection).

As we talked about in the sync-up, we should state that whether a union value can be null is determined by the union group's repetition: optional union groups can contain null and required union groups cannot. That way, we can always distinguish between a null value in the union (group level is null) and a non-null is a union option that wasn't projected.

Could you also update the logical types documentation?